docs/DeveloperGuide.md

```diff
@@ -666,152 +666,351 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
-      category.
+**Extensions:**
+- 1a. The specified company does not exist in the list of companies.
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+
+---
+
+#### Use Case: UC12 - Editing a Person
+
+**Preconditions:** User has the SOCareers App open and has added at least one person to their list of people.
+**Guarantees:** The specified person's details are updated with the provided details.
+
+**MSS:**
+1. User requests to edit a person.
+2. SOCareers updates the specified person's details with the provided details and displays confirmation in the UI.
+
+    Use case ends.
+
+**Extensions:**
+- 1a. The specified person does not exist in the list of people.
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+- 1b. The provided information is invalid (e.g., invalid phone number format, invalid email format, etc.).
+  - 1b1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+
+---
 
-  Use case resumes at step 2.
+#### Use Case: UC13 - Editing a Company
+
+**Preconditions:** User has the SOCareers App open and has added at least one company to their list of companies.
+**Guarantees:** The specified company's details are updated with the provided details.
+
+**MSS:**
+1. User requests to edit a company.
+2. SOCareers updates the specified company's details with the provided details and displays confirmation in the UI.
+
+    Use case ends.
+
+**Extensions:**
+- 1a. The specified company does not exist in the list of companies.
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+  - 1b. The provided information is invalid (e.g., invalid phone number format, invalid email format, etc.).
+    - 1b1. SOCareers displays an error message.
+
+      Use case resumes from step 1.
+
+---
+
+#### Use Case: UC14 - Editing an Internship
+
+**Preconditions:** User has the SOCareers App open and has added at least one company that has at least one internship.
+**Guarantees:** The specified internship's details are updated with the provided details.
+
+**MSS:**
+1. User requests to edit an internship.
+2. SOCareers updates the specified internship's details with the provided details and displays confirmation in the UI.
+
+    Use case ends.
+
+**Extensions:**
+- 1a. The specified company does not exist in the list of companies.
+  - 1a1. SOCareers displays an error message.
+
+    Use case resumes from step 1.
+- 1b. The specified internship does not exist in the specified company's list of internships.
+  - 1b1. SOCareers displays an error message.
+
+      Use case resumes from step 1.
+- 1c. The provided information is invalid (e.g., invalid datetime, etc.).
+  - 1c1. SOCareers displays an error message.
+
+      Use case resumes from step 1.
+
+---
+
+#### Use Case: UC15 - Finding People
+
+**Preconditions:** User has the SOCareers App open and has added at least one person to their list of people.
```

Comment:
Is `at least one` here needed for search? (Similar to UC05)

Same comment for UC16 and UC17

Reply:
Actually no. But I think the use cases can be stricter than the implementation?

Reply:
I will correct it.